stack/layout-of-doc-directory.rst

```diff
@@ -0,0 +1,200 @@
+############################
+Layout of the doc/ directory
+############################
+
+This page describes the layout of :file:`doc/` directories in packages.
+By following these guidelines, the documentation content of individual packages can successfully integrate with the build of `pipelines.lsst.io`_.
+
+.. _docdir-template:
+
+The stack_package template
+==========================
+
+Use the `stack_package template`_, and its examples, as references for implementing the :file:`doc/` directory of your package.
+This page will refer to those templates frequently.
+
+The :file:`doc/` directory structure of the example package looks like this:
+
+.. code-block:: text
+
+   doc
+   ├── SConscript
+   ├── conf.py
+   ├── doxygen.conf.in
+   ├── example
+   │   └── index.rst
+   ├── index.rst
+   ├── lsst.example
+   │   └── index.rst
+   └── manifest.yaml
+
+.. _docdir-manifest-yaml:
+
+The doc/manifest.yaml file
+==========================
+
+This YAML file describes the content of the package’s :file:`doc/` directory.
+The build process uses the :file:`doc/manifest.yaml` file to symlink documentation content from packages into the :ref:`main documentation repository <stack-docs-system-main-repo>`, `pipelines_lsst_io`_.
+
+The :file:`doc/manifest.yaml` file for a package named ``example`` looks like this:
+
+.. remote-code-block:: https://raw.githubusercontent.com/lsst/templates/master/project_templates/stack_package/example/doc/manifest.yaml
+   :language: yaml
+
+.. _docdir-manifest-yaml-package:
+
+package field
+-------------
+
+The ``package`` field must match the package’s EUPS name.
+This field implies the existence of a subdirectory named after the package (in the above example, the directory is :file:`doc/example`).
+This directory is called the :ref:`package documentation directory <docdir-package-doc-directory>` and provides the homepage for the package in the `pipelines.lsst.io`_ site.
+
+.. _docdir-manifest-yaml-modules:
+
+modules field
+-------------
+
+The ``modules`` field is a **list** of the public Python modules that the package provides.
+For most packages, as is the case for the example above, the package provides just one public Python module (``lsst.example``, for example).
+
+Each item in the ``modules`` list corresponds to a :ref:`module documentation directory <docdir-module-doc-directories>`.
+
+.. note::
+
+   “Modules” here means the major public namespace provided by a package, *not* the individual Python files in a package.
```

Comment:
You may want to additionally note that this is _different_ from standard Python terminology, where the individual files are actually modules and the directory with `__init__.py` is a package.